### PR TITLE
add LV_CHART_MODE_CIRCULAR feature to chart objects.

### PR DESCRIPTION
**Pull request by ali-rostami**

### CODE REVIEW

**Comment by ali-rostami** [comment]
there are things that I wasn't sure about. like:
I don't know if this check ```if(ser->points[p_prev] != LV_CHART_POINT_DEF && ser->points[p_act] != LV_CHART_POINT_DEF)``` is necessary in ```lv_chart_inv_lines``` function. As it is used in ```lv_chart_draw_lines```.
we need to talk about them, but before this, I thought it would be better to send the PR first and then talk about them here. so let's do it.

**Comment by embeddedt** [comment]
(cc @kisvegabor)

**Comment by kisvegabor** [comment]
@ali-rostami sorry for the delay. I've just realized that I didn't finish my review and you can't see it so far.

File: src/lv_objx/lv_chart.c [hunk 1]
```diff
@@ -380,12 +382,22 @@ void lv_chart_set_points(lv_obj_t * chart, lv_chart_series_t * ser, lv_coord_t *
...
-    ser->start_point = (ser->start_point + 1) % ext->point_cnt;
-
-    lv_chart_refresh(chart);
+	lv_chart_ext_t * ext = lv_obj_get_ext_attr(chart);
+	if(ext->update_mode == LV_CHART_MODE_SHIFT) {
+		ser->points[ser->start_point] = y;  /*This was the place of the former left most value, after shifting it is the rightmost*/
+		ser->start_point = (ser->start_point + 1) % ext->point_cnt;
+		lv_chart_refresh(chart);
+	} else if(ext->update_mode == LV_CHART_MODE_CIRCULAR) {
+		static uint16_t x = 0;
```
[the remark this answers is not in the document]
Comment (ali-rostami) [reply src/lv_objx/lv_chart.c]: yeah, you are right. we need a variable for every series. I'm not sure about the start_point. unfortunately, it is not possible for me to test it for about 2 weeks. can anybody do it?

**Comment by kisvegabor** [comment]
@ali-rostami 
I've done some simple tests and fixed a few things. See the last commit.

As the chart object is getting more complex a better test would very useful to verify UPDATE and SHIFT modes with all chart types.

Could you add a test like this as `lv_test_chart2` [here](https://github.com/littlevgl/lv_examples/blob/master/lv_tests/lv_test_objx/lv_test_chart/lv_test_chart.c)?

**Comment by ali-rostami** [comment]
So good thanks.
As I'm on a trip I can't promise to do it soon.-maybe for 2 weeks- But if you don't mind I will do it as soon as possible I can.

**Comment by kisvegabor** [comment]
No problem, have a nice trip! :)

**Comment by kisvegabor** [comment]
@ali-rostami Welcome back, I hope you had a nice trip! :smile: 

There were some changes in `dev-6.0` and now there are conflicts. I will fix them later today.

**Comment by ali-rostami** [comment]
> @ali-rostami Welcome back, I hope you had a nice trip! 😄
> 
> There were some changes in `dev-6.0` and now there are conflicts. I will fix them later today.

Yeah, I'm back. thanks 👍

**Comment by ali-rostami** [comment]
Oh, by the way, would you please implement the ```lv_chart_inv_areas```, too. 😎
thanks.

**Comment by kisvegabor** [comment]
I fixed the conflicts. `lv_chart_inv_lines` should work for `LV_CHART_TYPE_AREA` too.

I hope I didn't mess up anything while solving the conflicts but please test it :)

**Comment by kisvegabor** [comment]
@ali-rostami Do you think it's ready to merge? It's fine from my side.

**Comment by ali-rostami** [comment]
I think it's better to wait a little more. I want to test it with dev6.0 and as you know I'm struggling with it. :(

**Comment by kisvegabor** [comment]
Okay :)

**Comment by kisvegabor** [comment]
@ali-rostami 
I saw that the last commit is related to the calendar and not chart. 
We can leave it here now but next time please open a new PR for unrelated updates to keep the PRs clear.

**Comment by ali-rostami** [comment]
I wanted to do this, but I don't know why it was added to this one.

**Comment by embeddedt** [comment]
@ali-rostami It's because you are making all your changes on the `master` branch. A better workflow could be to create a separate branch for the chart patch, and another branch for the calendar change.

**Comment by kisvegabor** [comment]
I suggest merging. It seems working and it would be better having it in `dev-6.0` to let people test the updates. 
We can have a new PR if updates are required.

**Comment by ali-rostami** [comment]
OK, do it.

**Comment by kisvegabor** [comment]
Merged!

**Pull request merged**